common/lib/capa/capa/capa_problem.py

```diff
@@ -789,7 +793,9 @@ def _extract_html(self, problemtree):  # private
             answervariable = None
             if problemid in self.correct_map:
                 pid = input_id
-                status = self.correct_map.get_correctness(pid)
+                if not self.has_saved_answers:
+                    status = self.correct_map.get_correctness(pid)
```

Comment:
@staubina Can you add a comment for this condition to help understand what we are doing here?